packages/back-end/src/jobs/proxyUpdate.ts

```diff
@@ -28,8 +28,10 @@ export default function addProxyUpdateJob(ag: Agenda) {
 
     const connection = await findSDKConnectionById(connectionId);
     if (!connection) return;
-    if (!connection.proxy.enabled) return;
-    if (!connection.proxy.host) return;
+    if (!IS_CLOUD && (!connection.proxy.enabled || !connection.proxy.host))
+      return;
+    // note: sseEnabled indicates that we are using Cloud Proxy behind the scenes.
+    if (IS_CLOUD && !connection.sseEnabled) return;
```

Comment:
This logic is starting to get confusing. It might make sense to have a function, e.g. `function shouldLookForSdkConnection(): boolean {}` and perhaps branch for `IS_CLOUD` so we can read cloud checks more clearly from self-hosted checks.

Reply:
Sure, added a helper function

Reply:
Nice, that's so much easier to read 🎉